File: e-commerce-api/models/Order.js
const mongoose = require('mongoose');

const orderSchema = mongoose.Schema({
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    items: [
        {
            product: {
                type: mongoose.Schema.Types.ObjectId,
                ref: 'Product',
                required: true,
            },
            quantity: {
                type: Number,
                required: true,
                default: 1,
            },
            price: {
                type: Number,
                required: true,
            },
        },
    ],
    totalAmount: {
        type: Number,
        required: true,
    },
    status: {
        type: String,
        enum: ['pending', 'complete', 'processing', 'shipped', 'delivered', 'cancelled'],
        default: 'pending',
    },
    shippingAddress: {
        street: String,
        city: String,
        country: String,
        postalCode: String,
    },
    riderLocation: {
        type: { type: String, default: 'Point' },
        coordinates: { type: [Number], default: [0, 0] },
    },
    createdAt: {
        type: Date,
        default: Date.now,
    }
});
orderSchema.index({ riderLocation: '2dsphere' });
const Order = mongoose.model('Order', orderSchema);

module.exports = Order;

File: e_commerce/lib/models/order.dart
class Order {
  final String id;
  final String user;
  final List<OrderItem> items;
  final double totalAmount;
  final String shippingAddress;
  final String status;

  Order({
    required this.id,
    required this.user,
    required this.items,
    required this.totalAmount,
    required this.shippingAddress,
    required this.status,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['_id'],
      user: json['user'],
      items: (json['items'] as List)
          .map((e) => OrderItem.fromJson(e))
          .toList(),
      totalAmount: (json['totalAmount']).toDouble(),
      shippingAddress: json['shippingAddress'],
      status: json['status'],
    );
  }
}

class OrderItem {
  final String product;
  final int quantity;
  final double price;

  OrderItem({
    required this.product,
    required this.quantity,
    required this.price,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      product: json['product'],
      quantity: json['quantity'],
      price: (json['price']).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "product": product,
      "quantity": quantity,
    };
  }
}